File: ABC350~/ABC409/b.py
a = int(input())
t = list(map(int, input().split()))
ans = 0
hash = {}

for i in t:
    hash[i] = hash.get(i, 0) + 1

print(hash)
for i in hash:
    if hash[i] >= i:
        ans = max(hash[i],ans)

print(ans)

File: ABC350~/ABC409/b.rb
n = gets.to_i
s = gets.split.map(&:to_i)
s.sort!
ans = 0


(1..n).each do |i|

  left = 0
  right = n-1

  while right - left >1
    mid = (left+right)/2
    if s[mid] >= i
      right = mid
    else
      left = mid
    end
  end

  if s[left]==i
    ans = left+1
  else
    ans 
  end
end